test/lib/failOnError.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var assert = require('assert');
+var AwsHelper = require('./../../lib/index');
+var simple = require('simple-mock');
+
+describe('AwsHelper.failOnError', function (done) {
+  afterEach(function () {
+    simple.restore();
+  });
+
+  it('should do nothing if there is no error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var spy = simple.spy(fakeContext, 'fail');
+    AwsHelper.failOnError(null, {}, fakeContext);
+    assert(!spy.called);
+    done();
+  });
+
+  it('should call context fail function if error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var fakeError = 'There is an error';
+    var failMock = simple.mock(fakeContext, 'fail');
```

Comment:
the reason we use mocks against the `aws-sdk` is to mock the "contract" the sdk method has. (*i.e. don't test dependencies*)
in *this* case you are *designing* the contract, therefore you should be (unit) testing it.